infrastructure/code/code_html.go

```diff
@@ -30,172 +31,157 @@ import (
 	"github.com/snyk/snyk-ls/domain/snyk"
 )
 
+type IgnoreDetail struct {
+	Label string
+	Value string
+}
+
+type DataFlowItem struct {
+	Number         int
+	FilePath       string
+	StartLine      int
+	EndLine        int
+	StartCharacter int
+	EndCharacter   int
+	FileName       string
+	Content        string
+	StartLineValue int
+}
+
+type ExampleCommit struct {
+	CommitURL string
+	IconSVG   template.HTML
+	RepoName  string
+	DiffHTML  template.HTML
+}
+
 //go:embed template/details.html
 var detailsHtmlTemplate string
 
-func replaceVariableInHtml(html string, variableName string, variableValue string) string {
-	return strings.ReplaceAll(html, fmt.Sprintf("${%s}", variableName), variableValue)
-}
+var globalTemplate *template.Template
 
-func getDataFlowHeadingHtml(issue snyk.CodeIssueData) string {
-	dataFlowCount := len(issue.DataFlow)
-	stepWord := "step"
+func init() {
+	funcMap := template.FuncMap{
+		"vendorIconSvg": getVendorIconSvg,
+		"repoName":      getRepoName,
+		"codeDiffHtml":  getCodeDiffHtml,
+		"trimCWEPrefix": trimCWEPrefix,
+		"idxMinusOne":   idxMinusOne,
+	}
 
-	if dataFlowCount > 1 {
-		stepWord += "s"
+	var err error
+	globalTemplate, err = template.New("code").Funcs(funcMap).Parse(detailsHtmlTemplate)
```

Comment:
Should we take the product name? E.g. `product.Code`?

Reply:
Do you mean replacing the hardcoded string "code" with `product.ProductCode` which is defined in the package `product`?

```go
const (
	ProductOpenSource           Product = "Snyk Open Source"
	ProductCode                 Product = "Snyk Code"
	ProductInfrastructureAsCode Product = "Snyk IaC"
	ProductContainer            Product = "Snyk Container"
	ProductUnknown              Product = ""
)
```

The string for the `New()` function is relevant when we are dealing with more than one template. Currently we have only one which represents the Code Issue template which was organised by `<articles />` to keep some semantics when building the layout. 

I'm not sure if it's worth importing the package `product` to get the string `"Snyk Code"`:

```go
template.New(string(product.ProductCode))
```

WDYT?

Reply:
It's ok, to import the package, I think. We should not run into dependencies, and using a specific constant is in my opinion cleaner. The alternative would be to define a constant, but I believe using the product.ProductCode constant would be clearer, as we then also have the reference that ties the template directly to the product.